Add note about why go mod replace is needed. (#3547)
dagger/dagger: docs/current/partials/_install-sdk-go.md

go mod edit -replace github.com/docker/docker=github.com/docker/docker@v20.10.3-0.20220414164044-61404de7df1a+incompatible
```

:::note

> The replace statement is currently needed due to one of Dagger's dependencies using a replace statement in their Go module. There is an issue tracking fixes to this [here](https://github.com/dagger/dagger/issues/3391).

Reviewer: ```suggestion
The `replace` statement is currently needed due to one of Dagger's dependencies using a `replace` statement in its Go module. Learn more in this [GitHub tracking issue](https://github.com/dagger/dagger/issues/3391).
```
